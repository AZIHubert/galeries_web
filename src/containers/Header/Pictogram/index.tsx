import * as React from 'react';
import { CSSTransition } from 'react-transition-group';
import styled from 'styled-components';

interface ContainerI {
  marginLeft?: number;
  marginRight?: number;
}

interface PictogramI {
  hoverPictogram: React.ComponentType;
  marginLeft?: number;
  marginRight?: number;
  pictogram: React.ComponentType;
}

const Container = styled.div<ContainerI>`
  background-color: transparent;
  border: none;
  cursor: pointer;
  height: 24px;
  margin: ${({
    marginLeft,
    marginRight,
  }) => (
    `0px ${marginRight}px 0px ${marginLeft}px`
  )};
  position: relative;
  width: 24px;
`;

Container.defaultProps = {
  marginLeft: 0,
  marginRight: 0,
};

const InnerContainer = styled.div`
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: absolute;
  right: 0;
  &.fade-enter {
    opacity: 0;
  }
  &.fade-enter-active {
    opacity: 1;
    transition: 500ms;
  }
  &.fade-exit {
    opacity: 1;
  }
  &.fade-exit-active {
    opacity: 0;
    transition: 500ms;
  }
`;

const Pictogram = ({
  hoverPictogram: HoverPictogramComponent,
  marginLeft = 0,
  marginRight = 0,
  pictogram: PictogramComponent,
}: PictogramI) => {
  const [hover, setHover] = React.useState<boolean>(false);
  return (
    <Container
      marginLeft={marginLeft}
      marginRight={marginRight}
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
    >
      <CSSTransition
        classNames='fade'
        in={hover}
        timeout={300}
        unmountOnExit
      >
        <InnerContainer>
          <HoverPictogramComponent />
        </InnerContainer>
      </CSSTransition>
      <CSSTransition
        classNames='fade'
        in={!hover}
        timeout={300}
        unmountOnExit
      >
        <InnerContainer>
          <PictogramComponent />
        </InnerContainer>
      </CSSTransition>
    </Container>
  );
};

export default Pictogram;
