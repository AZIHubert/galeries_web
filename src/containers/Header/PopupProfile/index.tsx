import * as React from 'react';

import Modal from '#components/Modal';
import defaultProfilePicture from '#ressources/svg/defaultProfilePicture.svg';

import ModalTicket from './ModalTicket';

const PopupProfile = () => {
  const [openModal, setOpenModal] = React.useState<boolean>(false);
  const [loading, setLoading] = React.useState<boolean>(false);
  const handleClose = () => {
    if (!loading) {
      setOpenModal(false);
    }
  };
  return (
    <>
      <div
        data-testid='popupProfil'
      >
        <button>
          <img
            alt='profile picture'
            src={defaultProfilePicture}
          />
          <p>
              userName
          </p>
          <p>
              see your profile
          </p>
        </button>
        <button>
          <img
            alt='edit your information pictogram'
          />
          <p>
              Edit your informations
          </p>
        </button>
        <button
          data-testid='buttonTicketModal'
          onClick={() => setOpenModal((nextState) => !nextState)}
        >
          <img
            alt='send a ticket pictogram'
          />
          <p>
              Share your opinion? Find a bug?
          </p>
          <p>
              send a ticket
          </p>
        </button>
      </div>
      <Modal
        open={openModal}
        handleClose={handleClose}
      >
        <ModalTicket
          loading={loading}
          setLoading={setLoading}
        />
      </Modal>
    </>
  );
};

export default PopupProfile;
