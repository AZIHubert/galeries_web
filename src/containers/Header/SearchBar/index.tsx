import * as React from 'react';

import search from '#ressources/svg/search.svg';

import {
  Container,
  Input,
  Logotype,
} from './styles';

const SearchBar = () => {
  const [searchBarIsFocused, setSearchBarIsFocused] = React.useState<boolean>(false);
  const searchInputRef = React.useRef<HTMLInputElement | null>(null);
  return (
    <Container
      focused={searchBarIsFocused}
      onClick={() => {
        if (searchInputRef.current) {
          searchInputRef.current.focus();
        }
      }}
    >
      <Logotype
        alt='search pictogram'
        focused={searchBarIsFocused}
        src={search}
      />
      <Input
        onBlur={() => setSearchBarIsFocused(false)}
        onFocus={() => setSearchBarIsFocused(true)}
        placeholder={searchBarIsFocused ? '' : 'search on Galeries'}
        ref={searchInputRef}
      />
    </Container>
  );
};

export default SearchBar;
