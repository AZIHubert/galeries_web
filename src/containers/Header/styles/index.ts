export { default as Container } from './Container';
export { default as HeaderPart } from './HeaderPart';
export { default as InnerContainer } from './InnerContainer';
export { default as Logo } from './Logo';
