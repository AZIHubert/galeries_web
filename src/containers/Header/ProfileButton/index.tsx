import * as React from 'react';
import styled from 'styled-components';
import { CSSTransition } from 'react-transition-group';

import defaultProfilePicture from '#ressources/svg/defaultProfilePicture.svg';

import PopupProfile from './PopupProfile';

const Container = styled.div`
  position: relative;
`;

const Button = styled.button`
  align-items: center;
  background-color: transparent;
  border: none;
  border-left: ${({ theme }) => (
    `1px solid ${theme.colors.primary}`
  )};
  border-right: ${({ theme }) => (
    `1px solid ${theme.colors.primary}`
  )};
  color: ${({ theme }) => theme.colors.primary};
  cursor: pointer;
  display: flex;
  font-size: 1.1rem;
  padding: 0 30px;
  &:focus {
    outline: none;
  }
`;

const ProfileButtonImage = styled.img`
  border: ${({ theme }) => (
    `2px solid ${theme.colors.primary}`
  )};
  border-radius: 50%;
  height: 36px;
  margin-right: 20px;
  width: 36px;
`;

const ProfileButton = () => {
  const [openPopupProfil, setOpenPopupProfile] = React.useState<boolean>(false);
  const containerRef = React.useRef<HTMLDivElement | null>(null);
  const handleClickOutside = (event: any) => {
    if (containerRef.current && !containerRef.current.contains(event.target)) {
      setOpenPopupProfile(false);
    }
  };
  React.useEffect(() => {
    document.addEventListener('click', handleClickOutside, true);
    return () => {
      document.removeEventListener('click', handleClickOutside, true);
    };
  });
  return (
    <Container
      ref={containerRef}
    >
      <Button
        data-testid='profilButton'
        onClick={() => setOpenPopupProfile((lastState) => !lastState)}
      >
        <ProfileButtonImage
          alt='profile picture'
          src={defaultProfilePicture}
        />
          userName
      </Button>
      <CSSTransition
        classNames='fade'
        in={openPopupProfil}
        timeout={500}
        unmountOnExit
      >
        <PopupProfile />
      </CSSTransition>
    </Container>
  );
};

export default ProfileButton;
