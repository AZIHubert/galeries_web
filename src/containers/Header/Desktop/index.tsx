import * as React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  useHistory,
  useLocation,
} from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';

import Button from '#components/Button';
import Modal from '#components/Modal';

import logo from '#ressources/svg/logoG.svg';
import {
  CreateGalerie,
  CreateGalerieHover,
  Home,
  HomeHover,
  Notification,
  NotificationHover,
} from '#ressources/svgComponents';

import { fetchLogout } from '#store/actions';
import { userSelector } from '#store/selectors';

import CreateGalerieModal from './CreateGalerieModal';
import Pictogram from './Pictogram';
import ProfileButton from './ProfileButton';
import {
  Container,
  Fader,
  HeaderPart,
  InnerContainer,
  Logo,
  StyledLink,
} from './styles';

const Header = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const user = useSelector(userSelector);
  const location = useLocation();
  const handleClickHome = () => history.push('/');
  const [openCreateGalerie, setOpenCreateGalerie] = React.useState<boolean>(false);

  const handleOpenCreateGalerie = () => setOpenCreateGalerie(true);
  const handleCloseCreateGalerie = () => setOpenCreateGalerie(false);

  const show = !!user
    && !location.pathname.includes('profilePicture')
    && !location.pathname.includes('updateEmail')
    && !location.pathname.includes('validateEmail');

  return (
    <CSSTransition
      classNames='fade'
      in={show}
      timeout={300}
      unmountOnExit
    >
      <Fader>
        <Container>
          <InnerContainer>
            <HeaderPart>
              <StyledLink
                to='/'
              >
                <Logo
                  alt="header logo"
                  src={logo}
                />
              </StyledLink>
            </HeaderPart>
            <HeaderPart>
              <Pictogram
                hoverPictogram={HomeHover}
                marginRight={25}
                marginRightL={35}
                onClick={handleClickHome}
                pictogram={Home}
              />
              <Pictogram
                hoverPictogram={CreateGalerieHover}
                marginRight={25}
                marginRightL={35}
                onClick={handleOpenCreateGalerie}
                pictogram={CreateGalerie}
              />
              <Pictogram
                hoverPictogram={NotificationHover}
                pictogram={Notification}
              />
              <ProfileButton />
              <Button.Header
                onClick={() => dispatch(fetchLogout())}
                small
                styles={{
                  marginLeft: 40,
                }}
                title='logout'
              />
            </HeaderPart>
          </InnerContainer>
          <Modal.Portal
            handleClose={handleCloseCreateGalerie}
            open={openCreateGalerie}
          >
            <CreateGalerieModal
              handleCloseCreateGalerie={handleCloseCreateGalerie}
            />
          </Modal.Portal>
        </Container>
      </Fader>
    </CSSTransition>
  );
};

export default Header;
