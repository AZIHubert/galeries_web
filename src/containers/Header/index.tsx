import * as React from 'react';
import {
  useDispatch,
} from 'react-redux';

import HeaderButton from '#components/HeaderButton';

import logo from '#ressources/svg/logoG.svg';
import {
  CreateGalerie,
  CreateGalerieHover,
  Home,
  HomeHover,
  Notification,
  NotificationHover,
} from '#ressources/svgComponents';

import { fetchLogout } from '#store/actions';

import Pictogram from './Pictogram';
import ProfileButton from './ProfileButton';
import SearchBar from './SearchBar';
import {
  Container,
  HeaderPart,
  InnerContainer,
  Logo,
} from './styles';

const Header = () => {
  const dispatch = useDispatch();
  return (
    <Container>
      <InnerContainer>
        <HeaderPart>
          <Logo
            src={logo}
            alt="header logo"
          />
          <SearchBar />
        </HeaderPart>
        <HeaderPart>
          <Pictogram
            hoverPictogram={HomeHover}
            pictogram={Home}
            marginRight={25}
            marginRightL={35}
          />
          <Pictogram
            hoverPictogram={CreateGalerieHover}
            pictogram={CreateGalerie}
            marginRight={25}
            marginRightL={35}
          />
          <Pictogram
            hoverPictogram={NotificationHover}
            pictogram={Notification}
          />
          <ProfileButton />
          <HeaderButton
            marginLeft={40}
            onClick={() => dispatch(fetchLogout())}
            small
            title='logout'
          />
        </HeaderPart>
      </InnerContainer>
    </Container>
  );
};

export default Header;
