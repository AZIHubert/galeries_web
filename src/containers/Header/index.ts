import Desktop from './Desktop';
import Home from './Home';

const Header = {
  Desktop,
  Home,
};

export default Header;
