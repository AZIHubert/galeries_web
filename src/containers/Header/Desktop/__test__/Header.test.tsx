import { cleanup } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import ReactRouterDOM from 'react-router-dom';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';

import ThemeProvider from '#contexts/ThemeContext';
import reducers from '#store/reducers';

import Header from '../index';

const mockedStore = createStore(reducers);

const Container = () => (
  <Provider store={mockedStore}>
    <ThemeProvider>
      <Header />
    </ThemeProvider>
  </Provider>
);

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as typeof ReactRouterDOM,
  Link: jest.fn(({ children }) => children),
  useLocation: () => ({
    pathname: 'localhost:3000/example/path',
  }),
}));

describe('Header', () => {
  afterEach(cleanup);
  describe('Desktop', () => {
    it('renders without crashing', () => {
      const tree = renderer.create(<Container />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
