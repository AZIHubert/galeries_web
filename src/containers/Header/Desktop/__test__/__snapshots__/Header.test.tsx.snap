// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header
  className="sc-cOajty dJwOpu"
>
  <div
    className="sc-khAkjo fmwWGD"
  >
    <div
      className="sc-AzgDb kpRfcw"
    >
      <img
        alt="header logo"
        className="sc-hTZhsR kyweWU"
        src={Object {}}
      />
      <div
        className="sc-jNMdTA bueqA-D"
        onClick={[Function]}
      >
        <img
          alt="search pictogram"
          className="sc-bBrOnJ dPLKca"
          src={Object {}}
        />
        <input
          className="sc-dOSReg ievEAf"
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="search on Galeries"
        />
      </div>
    </div>
    <div
      className="sc-AzgDb kpRfcw"
    >
      <div
        className="sc-bqyKva bOOLZl"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          onClick={[Function]}
        >
          <div
            className="sc-hBEYos ggqouw"
          >
            <svg
              className="sc-jrAGrp xQKQQ"
              viewBox="0 0 784.16 904.91"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
              >
                <path
                  d="M784.16 904.91H450.91V665.59H332.57v239.32H0V336.39L392.08 0l392.08 336.39zm-258.25-75h183.25v-459l-317.08-272L75 370.86v459.05h182.57V590.59h268.34z"
                  data-name="Layer 1"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-bqyKva bOOLZl"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          onClick={[Function]}
        >
          <div
            className="sc-hBEYos ggqouw"
          >
            <svg
              className="sc-iBPRYJ iBHXVy"
              viewBox="0 0 893 893"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
                id="createGalerie_svg__Layer_2"
              >
                <g
                  data-name="Layer 1"
                  id="createGalerie_svg__Layer_1-2"
                >
                  <path
                    className="createGalerie_svg__cls-1"
                    d="M711.5 0h-530C81.42 0 0 81.42 0 181.5v530C0 811.58 81.42 893 181.5 893h530C811.58 893 893 811.58 893 711.5v-530C893 81.42 811.58 0 711.5 0zM818 711.5A106.63 106.63 0 01711.5 818h-530A106.63 106.63 0 0175 711.5v-530A106.63 106.63 0 01181.5 75h530A106.63 106.63 0 01818 181.5z"
                  />
                  <path
                    className="createGalerie_svg__cls-1"
                    d="M484 222.94h-75V409H222.94v75H409v186.06h75V484h186.06v-75H484V222.94z"
                  />
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-bqyKva LLOXI"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          onClick={[Function]}
        >
          <div
            className="sc-hBEYos ggqouw"
          >
            <svg
              className="sc-iqHYGH gpBMoq"
              viewBox="0 0 998.13 907.88"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
              >
                <path
                  d="M498.35 907.88L91.66 501.19l1.42-1.42-8.85-8.85A287.57 287.57 0 11490.92 84.23l8.85 8.85 7.44-7.44a287.59 287.59 0 01406.69 0 287.58 287.58 0 010 406.7zM197.73 501.19l300.63 300.63 362.51-362.51c82.88-82.89 82.88-217.75 0-300.63s-217.75-82.89-300.63 0l-60.47 60.47-61.88-61.89c-82.88-82.88-217.75-82.88-300.63 0a212.58 212.58 0 000 300.63l61.89 61.88z"
                  data-name="Layer 1"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        className="sc-eggNIi fCWDMX"
      >
        <button
          className="sc-iBaPrD izHVKr"
          onClick={[Function]}
        >
          <div
            className="sc-cTkwdZ jHPlsD"
          >
            <div
              className="sc-fodVxV bpNrRt"
            />
          </div>
          user name
        </button>
      </div>
      <button
        className="sc-eCssSg goafQX"
        onClick={[Function]}
      >
        logout
      </button>
    </div>
  </div>
</header>
`;
